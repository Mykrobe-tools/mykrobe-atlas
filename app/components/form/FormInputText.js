/* @flow */

import React, { Component, PropTypes } from 'react';

import FormLabel from './FormLabel';

import styles from './FormInputText.css';

class FormInputText extends Component {

  render() {
    const {name, onChange, placeholder, title, type, value} = this.props;
    const initialValue = value || ''; /* don't pass undefined value https://github.com/facebook/react/issues/6222  */
    return (
      <div className={styles.wrap}>
        <div className={styles.label}>
          <FormLabel
            htmlFor={name}
            label={title} />
        </div>
        <input
          className={styles.input}
          id={name}
          name={name}
          type={type}
          value={initialValue}
          placeholder={placeholder}
          onChange={(event) => onChange(event)} />
      </div>
    );
  }
}

FormInputText.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['text', 'number', 'email', 'tel', 'search', 'password']).isRequired,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
};

export default FormInputText;
