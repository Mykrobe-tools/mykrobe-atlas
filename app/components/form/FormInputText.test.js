import * as React from 'react';
import renderer from 'react-test-renderer';
import FormInputText from './FormInputText';

describe('FormInputText component snapshot', () => {
  const name = 'loremIpsum';
  const title = 'Lorem Ipsum';
  const type = 'text';
  const placeholder = 'Lorem Ipsum Dolor Sit Amet';
  const value = 'Dolor sit amet';
  const onChange = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <FormInputText
        name={name}
        title={title}
        value={value}
        placeholder={placeholder}
        type={type}
        onChange={onChange}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
