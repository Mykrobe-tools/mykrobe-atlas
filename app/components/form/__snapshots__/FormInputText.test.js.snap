// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputText component snapshot renders correctly 1`] = `
<div
  className="wrap"
>
  <div
    className="label"
  >
    <label
      className="label"
      htmlFor="loremIpsum"
    >
      Lorem Ipsum
    </label>
  </div>
  <input
    className="input"
    id="loremIpsum"
    name="loremIpsum"
    onChange={[Function]}
    placeholder="Lorem Ipsum Dolor Sit Amet"
    type="text"
    value="Dolor sit amet"
  />
</div>
`;
